keras/layers/embeddings.py

```diff
@@ -125,6 +125,7 @@ def get_output_shape_for(self, input_shape):
         return (input_shape[0], input_length, self.output_dim)
 
     def call(self, x, mask=None):
+        x = K.cast(x, 'int32')
```

Comment:
For performance reasons, better to only add operations that are actually necessary, e.g.

```python
if K.dtype(x) != 'int32':
   x = K.cast(x, 'int32')
```